### PR TITLE
Multidimension support for nn.functional.normalize

### LINKED ISSUE #70100: dim parameter of torch.nn.functional.normalize should accept tuples
### 🐛 Describe the bug

torch.nn.functional.normalize only accepts ints for the dim parameter, however should work in the multidim case

The following code works as expected, however mypy will fail as `dim` is typed as an int.
```
import torch
import torch.nn.functional as F

F.normalize(torch.rand(2,3,4), dim=(0,1))
```

### Versions

Collecting environment information...
PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: 11.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.15.5
Libc version: glibc-2.27

Python version: 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:20:46)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-1064-azure-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: 10.1.243
GPU models and configuration:
GPU 0: Tesla V100-PCIE-16GB
GPU 1: Tesla V100-PCIE-16GB
GPU 2: Tesla V100-PCIE-16GB
GPU 3: Tesla V100-PCIE-16GB

Nvidia driver version: 460.91.03
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.6.3
[pip3] mypy==0.910
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.3
[pip3] pytorch-lightning==1.5.0
[pip3] segmentation-models-pytorch==0.2.0
[pip3] torch==1.10.0
[pip3] torchmetrics==0.6.0
[pip3] torchvision==0.11.1
[conda] cudatoolkit               11.5.0               h36ae40a_9    conda-forge
[conda] efficientnet-pytorch      0.6.3                    pypi_0    pypi
[conda] libblas                   3.9.0            12_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            12_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            12_linux64_mkl    conda-forge
[conda] magma                     2.5.4                h6103c52_2    conda-forge
[conda] mkl                       2021.4.0           h8d4b97c_729    conda-forge
[conda] mypy                      0.910                    pypi_0    pypi
[conda] mypy-extensions           0.4.3                    pypi_0    pypi
[conda] numpy                     1.21.3           py39hdbf815f_1    conda-forge
[conda] pytorch                   1.10.0          cuda112py39h4e14dd4_0    conda-forge
[conda] pytorch-gpu               1.10.0          cuda112py39h0bbbad9_0    conda-forge
[conda] pytorch-lightning         1.5.0                    pypi_0    pypi
[conda] segmentation-models-pytorch 0.2.0                    pypi_0    pypi
[conda] torchmetrics              0.6.0                    pypi_0    pypi
[conda] torchvision               0.11.1                   pypi_0    pypi

### PR DESCRIPTION
**Pull request by calebrob6**
Fixes #70100

### CODE REVIEW

**Comment by calebrob6** [comment]
I'm not sure what exactly is failing based on the test output. Any pointers?

File: torch/nn/functional.py [hunk 1]
```diff
@@ -4587,7 +4593,7 @@ def normalize(input: Tensor, p: float = 2.0, dim: int = 1, eps: float = 1e-12, o
     Args:
         input: input tensor of any shape
         p (float): the exponent value in the norm formulation. Default: 2
-        dim (int): the dimension to reduce. Default: 1
+        dim (Union[int, Sequence[int]]): the dimensions to reduce over. Default: 1
```
Comment (jbschlosser) [inline torch/nn/functional.py:4596]: nit: rather than specifying the formal type here, I think this should follow the wording used by the [`torch.norm()` docs](https://github.com/pytorch/pytorch/blob/6623c4838ede2f04f4708a4335dc3668c97523c1/torch/functional.py#L1484):
```
dim (int, tuple of ints, list of ints): the dimensions to reduce over. Default: 1
```

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! Just one tiny nitpicky thing for consistency with other docs but otherwise looks good to go :)

**Comment by calebrob6** [comment]
Thanks @jbschlosser, are the test failures of concern?

**Comment by jbschlosser** [comment]
> Thanks @jbschlosser, are the test failures of concern?

I don't see any failures as of now - should be fine as long as mypy passes.

**Review by jbschlosser — APPROVED** [review]
LGTM!

**Comment by jbschlosser** [comment]
> Thanks @jbschlosser, are the test failures of concern?

Whoops spoke too soon - the change caused quite a few JIT-related failures that can't be ignored. @calebrob6

**Comment by nps1ngh** [comment]
@jbschlosser @calebrob6 

The problem seems to be that `Tensor.norm`[^1] wants `dim` to be `List[int]` and which the supplied type hint `Union[int, Sequence[int]]` doesn't support.
There are two things that need to be done here to satify the JIT compiler:
1. Convert `dim` to a list if it isn't.
2. Change `Sequence[int]` to `List[int]`. (It does not work otherwise.)

So something along the lines of[^2]:

```python
@torch.jit.script
def normalize(x, dim: Union[int, List[int]]):
    if not isinstance(dim, list):
        dim = [dim]
    return x / x.norm(2, dim=dim, keepdim=True).clamp_min(1e12)
```

On torch 1.13.


[^1]: Also, `torch.linalg.vector_norm` causes the same issue as `Tensor.norm` (I suppose they dispatch to the same, hence why).
[^2]: The `expand_as` seems redundant to me due to broadcasting, or did I overlook something?
